src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/Views/DynamicTableViewConsentCell.swift

```diff
@@ -0,0 +1,184 @@
+			consentStackView.addArrangedSubview($0)
+		}
+		consentStackView.setCustomSpacing(16, after: seperatorView1)
+		consentStackView.setCustomSpacing(16, after: flagCountriesLabel)
+		consentStackView.setNeedsUpdateConstraints()
+	}
+	
+	
+	private func setupConstraints() {
+		let marginGuide = contentView.layoutMarginsGuide
+		
+
+		NSLayoutConstraint.activate([
+			consentView.leadingAnchor.constraint(equalTo: marginGuide.leadingAnchor),
+			consentView.topAnchor.constraint(equalTo: marginGuide.topAnchor),
+			consentView.trailingAnchor.constraint(equalTo: marginGuide.trailingAnchor),
+			consentView.bottomAnchor.constraint(equalTo: marginGuide.bottomAnchor),
+			consentStackView.topAnchor.constraint(equalTo: consentView.topAnchor, constant: 20),
+			consentStackView.trailingAnchor.constraint(equalTo: consentView.trailingAnchor, constant: -16),
+			consentStackView.leadingAnchor.constraint(equalTo: consentView.leadingAnchor, constant: 16),
+			consentStackView.bottomAnchor.constraint(equalTo: consentView.bottomAnchor, constant: -20),
+			seperatorView1.heightAnchor.constraint(equalToConstant: 1),
+			seperatorView2.heightAnchor.constraint(equalToConstant: 1)
+
+		])
+		
+	}
+	
+	func configure(
+		subTitleLabel: NSMutableAttributedString,
+		descriptionPart1Label: NSMutableAttributedString,
+		descriptionPart2Label: NSMutableAttributedString,
+		countries: [Country],
+		descriptionPart3Label: NSMutableAttributedString,
+		descriptionPart4Label: NSMutableAttributedString
+	) {
+		self.subTitleLabel.attributedText = subTitleLabel
+		self.descriptionPart1Label.attributedText = descriptionPart1Label
+		self.descriptionPart2Label.attributedText = descriptionPart2Label
+		self.descriptionPart3Label.attributedText = descriptionPart3Label
+		self.descriptionPart4Label.attributedText = descriptionPart4Label
+		self.flagCountriesLabel.text = countries.map { $0.localizedName }.joined(separator: ", ")
+		
+		let flagString = NSMutableAttributedString()
+		
+		countries
+			.compactMap { $0.flag?.withRenderingMode(.alwaysOriginal) }
+			.forEach { flag in
+				let imageAttachment = NSTextAttachment()
+				imageAttachment.image = flag
+				imageAttachment.setImageHeight(height: 17)
+				let imageString = NSAttributedString(attachment: imageAttachment)
+				flagString.append(imageString)
+				flagString.append(NSAttributedString(string: "   "))
+			}
+
+		let style = NSMutableParagraphStyle()
+		style.lineSpacing = 10
+		flagString.addAttribute(.paragraphStyle, value: style, range: NSRange(location: 0, length: flagString.length))
+				
+		self.flagIconsLabel.attributedText = flagString
+	}
+}
+
+extension NSTextAttachment {
+	func setImageHeight(height: CGFloat) {
+		guard let image = image else { return }
+		let ratio = image.size.width / image.size.height
+
+		bounds = CGRect(x: bounds.origin.x, y: bounds.origin.y, width: ratio * height, height: height)
+	}
+}
```

Comment:
Do we have an extra file for NSTestAttachments yet? If so, move it over there overwise create a new file, so others can finde this more easy

Reply:
Done. Move into the extension classes folder.